android/guava/src/com/google/common/cache/CacheBuilder.java

```diff
@@ -231,13 +231,16 @@ public CacheStats snapshot() {
           });
   static final CacheStats EMPTY_STATS = new CacheStats(0, 0, 0, 0, 0, 0);
 
-  static final Supplier<StatsCounter> CACHE_STATS_COUNTER =
-      new Supplier<StatsCounter>() {
-        @Override
-        public StatsCounter get() {
-          return new SimpleStatsCounter();
-        }
-      };
+  /*
+   * We avoid using a method reference here for now: Inside Google, CacheBuilder is used from the
+   * implementation of a custom ClassLoader that is sometimes used as a system classloader. That's a
+   * problem because method-reference linking tries to look up the system classloader, and it fails
+   * because there isn't one yet.
+   *
+   * I would have guessed that a lambda would produce the same problem, but maybe it's safe because
+   * the lambda implementation is generated as a method in the _same class_ as the usage?
+   */
+  static final Supplier<StatsCounter> CACHE_STATS_COUNTER = () -> new SimpleStatsCounter();
```

Comment:
**Scalability Issue**: Changing from an anonymous class to a lambda for the 'CACHE_STATS_COUNTER' initialization could potentially impact scalability, depending on how the JVM optimizes lambda expressions vs. anonymous classes. Lambdas in Java 8 are generally more efficient than anonymous classes, especially when they are stateless or effectively final, as they can be optimized better by the JVM. <br> **Fix**: The change to use a lambda expression is likely beneficial for scalability. However, it's important to benchmark and monitor the performance impact of this change in environments where the cache is under high load, to ensure that the theoretical scalability improvements are realized in practice. <br> **Code Suggestion**: 
 ```
 -      new Supplier<StatsCounter>() {
 -        @Override
 -        public StatsCounter get() {
 -          return new SimpleStatsCounter();
 -        }
 -      };
 +  /*
 +   * We avoid using a method reference here for now: Inside Google, CacheBuilder is used from the
 +   * implementation of a custom ClassLoader that is sometimes used as a system classloader. That's a
 +   * problem because method-reference linking tries to look up the system classloader, and it fails
 +   * because there isn't one yet.
 +   *
 +   * I would have guessed that a lambda would produce the same problem, but maybe it's safe because
 +   * the lambda implementation is generated as a method in the _same class_ as the usage?
 +   */
 +  static final Supplier<StatsCounter> CACHE_STATS_COUNTER = () -> new SimpleStatsCounter();
 ```